conbench/entities/history.py

```diff
@@ -570,3 +586,77 @@ def _calculate_z_score(
         z_score = z_score * -1
 
     return z_score
+
+
+def _detect_shifts_with_trimmed_estimators(
+    df: pd.DataFrame, z_score_threshold=5.0
```

Comment:
```suggestion
    df: pd.DataFrame, z_score_threshold=10.0
```

I know 5 is used a bunch of places, but IMO we should increase this default (cf https://github.com/voltrondata-labs/arrow-benchmarks-ci/pull/100). Is this used anywhere? If it isn't (and we're using whatever the config value is) let's not dig the 5 hole deeper and make this be the default we want (bonus points for going through and undoing that default elsewhere, can be a separate PR)

Reply:
I don't actually agree here; as @austin3dickey said in https://github.com/conbench/conbench/pull/956#discussion_r1149668922 this is not quite the same as the config value:

> oof, yeah, it's pretty different from the existing "z-score threshold" concept in this codebase, because it operates on a (clipped) mean-diff distribution instead of a lookback mean distribution. Maybe this parameter could be something like "mean_diff_z_score_threshold" or something less bad than that? ;)
>
> I think it's fine to not introduce the server-wide configuration environment variable until someone asks for it later.

Because of the trimming, this doesn't have the volatility problems we have elsewhere with 5%. Raising it to 10% will result in ignoring a lot of obvious outliers. If we're going to use the plots to keep an eye on how this is performing, this is exactly the lever we can pull to adjust if it's over- or under-identifying outliers. Making it more conservative now means we won't see the line of what it considers an outlier as much.